### PR TITLE
#137 - Splitting TopSectionFinder from Renderer and adding unit tests to both. (WIP)

### PR DESCRIPTION
**Pull request by brandongk60**
## Issue:
#137 

Splitting up the renderer.ts file to facilitate testing.

### CODE REVIEW

File: src/renderer.ts [hunk 1]
```diff
@@ -20,25 +15,22 @@ export interface ICoverageLines {
...
+        this.topSectionFinder = topSectionFinder;
     }
 
     /**
      * Renders coverage to editors
      * @param sections cached set of sections
      * @param textEditors currently visible text editors
      */
-    public async renderCoverage(
+    public renderCoverage(
```
Comment (brandongk60) [inline src/renderer.ts]: @ryanluker - I don't think this function needs to be `async`...

File: src/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export function normalizeFileName(fileName: string): string {
+    let name = fileName;
+    // make file path relative and OS independent
+    name = name.toLocaleLowerCase();
+    // remove all file slashes
+    name = name.replace(/\//g, "###");
+    name = name.replace(/\\/g, "###");
+    return name;
+}
```
Comment (ryanluker) [inline src/helpers.ts]: good idea, helper library makes sense at this point!

File: src/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    const subInt = intersection.reverse().join("");
+    return subInt;
+}
+
+export function normalizeFileName(fileName: string): string {
+    let name = fileName;
+    // make file path relative and OS independent
+    name = name.toLocaleLowerCase();
+    // remove all file slashes
```
Comment (ryanluker) [inline src/helpers.ts]: wasnt sure of a node api that would properly strip unix and posix slashes to allow for comparisons... the use case here is windows 10 allows for WSL to generate coverage and have a the extension consume it in windows.

File: src/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    const intersection: string[] = [];
+    let pos = 0;
+    // stop when strings at pos are no longer are equal
+    while (a[pos] === b[pos]) {
+        // if we reached the end or there isnt a value for that pos
+        // exit the while loop
+        if (!a[pos] || !b[pos]) { break; }
+        intersection.push(a[pos]);
+        pos++;
+    }
```
Comment (ryanluker) [inline src/helpers.ts]: this while loop is pretty dangerous and has caused a few bugs. originally I was considering recursion here with a exit condition of getting to the end of one array. it is fine the way it is though as I see you have quite a few more tests backing it thanks!

File: src/renderer.ts (hunk 1 above)
Comment (ryanluker) [reply src/renderer.ts]: yep doesnt look that way good catch!

**Pull request merged**